src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -38,6 +38,9 @@ abstract class AbstractRowBuffer<T> implements RowBuffer<T> {
   // side scanner
   private static final int INVALID_SERVER_SIDE_DATA_TYPE_ORDINAL = -1;
 
+  // Maximum number of rows we recommend to pass to insertRows()
+  private static final int RECOMMENDED_MAX_BATCH_SIZE = 10_000;
```

Comment:
Worth making a config option?

Reply:
Created an option

Reply:
This is fine, but I don't think we should keep adding configurable constants (at least for the warning one)